### PR TITLE
♻️ Refactor/debt

### PR DESCRIPTION
**Pull request by Thomas-Smets**
First stage of the refactor: We keep legacy lending functionality, but we shadow every interaction related to lending in the new lending repo. After every interaction we check that the total amount of debt in lending legacy equals total amount of debt in new lending repo (last line in syncDebt() in Vault.sol.

If all tests succeed, we know that new lending repo behaves as expected.

Next two steps of the refactor:
2) Replace all legacy functions with logic from new lending repo, remove stable contracts -> rewrite all tests calling stable.sol
3) Remove all lending functionality from the vault -> Rewrite all tests to take out/repay loans from arcadia-lending instead from the vaults

### CODE REVIEW

**Comment by Thomas-Smets** [comment]
Wait with reviewing, I'll give some extra info in the afternoon

File: src/test/14_ATokenSubregistry.t.sol [hunk 1]
```diff
@@ -59,7 +59,8 @@ contract aTokenSubRegistryTest is Test {
                 baseCurrencyToUsdOracleUnit: 0,
                 assetAddress: 0x0000000000000000000000000000000000000000,
                 baseCurrencyToUsdOracle: 0x0000000000000000000000000000000000000000,
-                stableAddress: 0x0000000000000000000000000000000000000000,
+                liquidityPool: 0x0000000000000000000000000000000000000000,
+stable: 0x0000000000000000000000000000000000000000,
```
Comment (zekiblue) [inline src/test/14_ATokenSubregistry.t.sol:63]: indention typo?

File: src/test/14_ATokenSubregistry.t.sol [hunk 2]
```diff
@@ -98,7 +99,8 @@ contract aTokenSubRegistryTest is Test {
...
```
Comment (zekiblue) [inline src/test/14_ATokenSubregistry.t.sol:103]: indention i guess

File: src/test/14_ATokenSubregistry.t.sol [hunk 3]
```diff
@@ -110,7 +112,8 @@ contract aTokenSubRegistryTest is Test {
                 ),
                 assetAddress: address(eth),
                 baseCurrencyToUsdOracle: address(oracleEthToUsd),
-                stableAddress: 0x0000000000000000000000000000000000000000,
+                liquidityPool: 0x0000000000000000000000000000000000000000,
+stable: 0x0000000000000000000000000000000000000000,
```
Comment (zekiblue) [inline src/test/14_ATokenSubregistry.t.sol:116]: indention

**Review by zekiblue — APPROVED** [review]

**Pull request merged**